src/Reflex/Dom/ACE.hs

```diff
@@ -143,233 +142,151 @@ data AceDynConfig = AceDynConfig
 instance Default AceConfig where
     def = AceConfig def def def False False
 
-#ifndef ghcjs_HOST_OS
-data Element = Element
-data JSVal = JSVal
-jsNull :: JSVal
-jsNull = JSVal
-
-jsval :: a -> JSVal
-jsval = const JSVal
-
-toJSString :: a -> b
-toJSString _ = undefined
-
-#endif
-
-newtype AceRef = AceRef { unAceRef :: JSVal }
+newtype AceInstance = AceInstance { unAceInstance :: JSVal }
 
 data ACE t = ACE
-    { aceRef   :: Dynamic t (Maybe AceRef)
+    { aceRef   :: Dynamic t (Maybe AceInstance)
     , aceValue :: Dynamic t Text
     }
 
-mtext2val :: Maybe Text -> JSVal
-mtext2val = maybe jsNull (jsval . toJSString)
+mtext2val :: Maybe Text -> JSM JSVal
+mtext2val = maybe (pure jsNull) (ghcjsPure . jsval . toJSString)
```

Comment:
Yup, at least in ghcjs https://github.com/ghcjs/ghcjs-base/blob/master/GHCJS/Marshal.hs#L257-L259
Hmm if `jsaddle` in ghc has a different behavior it is a bug that should be fixed :)